specs/toggle.md

```diff
@@ -0,0 +1,92 @@
+# fast-toggle
+
+## Overview
+
+*Toggle* mimics a physical switch that allows users to turn things **on** or **off**.
+
+### Use Cases
+
+*Toggle* is used to represent a binary state such and **on** or **off**. If any extra step is required for changes to be effective or an indeterminate state is needed, you should use *checkbox* and corresponding "Apply" button instead. Users shouldn’t need to do something else or go somewhere else in order to experience the *Toggle's* effect.
+
+Typical use cases include, but are not limited to turning a feature on and off or showing or hiding a piece of UI
+  
+### Features
+
+A toggle should allow the following properties. Selected, whether the toggle is selected or not. Selected message, the message that displays when the toggle is selected. Unselected message, the message that displays when toggle is not selected. Disabled, whether the toggle is disabled or not.
+
+### Risks and Challenges
+
+Typically it's best to include a selected and unselected message but it is not required. It's common to see *Toggle* without a message but there are accessibility concerns since it's not necessarily obvious what is **on** and what is **off**. The visual representation of **on** tends to show the state indicator to the right with the background filled and **off** tends to show the state indicator to the left and the background unfilled. However, this pattern is not always consistent.
+
+## Design
+
+*Describe the design of the component, thinking through several perspectives:*
+
+- *A customer using the component on a web page.*
+- *A developer building an app with the component and interacting through HTML/CSS/JavaScript.*
+- *A designer customizing the component.*
+
+### API
+
+- fast-toggle
+- selected, unselected message, selected message, disabled, labelId, statusMessageId
+- onChange
+
+### Anatomy and Appearance
+
+<div>
+    <label id="toggle-label">Toggle label</label>
+    <button role="switch" aria-checked="true" aria-labelledby="toggle-label" aria-describedby="status-message">
+        <!-- Status Indicator -->
+        <span/>
```

Comment:
How will an app author style these component parts? Which [`::part`](https://developer.mozilla.org/en-US/docs/Web/CSS/::part)s does it support?

Reply:
@eljefe223 In case you need a reference on `::part`, here's some MDN docs: [CSS Shadow Parts](https://developer.mozilla.org/en-US/docs/Web/CSS/::part) We can basically put a `part` attribute on any HTML element in our Shadow DOM and it will enable the user to select that part with a `::part` pseudo-element selector for more style customization.

Reply:
Here's another helpful link to the part explainer: https://meowni.ca/posts/part-theme-explainer/ I'll try to add some of these to the spec template for future reference.